llpc/context/llpcCompiler.cpp

```diff
@@ -1418,12 +1429,162 @@ Result Compiler::buildGraphicsPipelineInternal(GraphicsContext *graphicsContext,
   }
 
   if (result != Result::Success) {
-    result = buildPipelineInternal(context, shaderInfo, /*unlinked=*/false, pipelineElf, stageCacheAccesses);
+    // See if we want to use part-pipeline compilation, if there is an FS and at least one pre-rasterization
+    // shader stage.
+    unsigned stageMask = context->getShaderStageMask();
+    bool buildPartPipeline = (cl::EnablePartPipeline && (stageMask & (1 << ShaderStageFragment)) &&
+                              (stageMask & ~(1 << ShaderStageFragment)));
+    if (buildPartPipeline)
+      result = buildGraphicsPipelineWithPartPipelines(context, shaderInfo, pipelineElf, stageCacheAccesses);
+    else
+      result = buildPipelineInternal(context, shaderInfo, PipelineLink::WholePipeline, nullptr, pipelineElf, stageCacheAccesses);
   }
   releaseContext(context);
   return result;
 }
 
+// =====================================================================================================================
+// Build graphics pipeline using part-pipeline compilation
+//
+// @param graphicsContext : Graphics context this graphics pipeline
+// @param shaderInfo : Shader info of this graphics pipeline
+// @param [out] pipelineElf : Output Elf package
+// @param [out] stageCacheAccesses : Stage cache access result. All elements must be initialized in the caller as
+//                                   CacheAccessInfo::CacheNotChecked
```

Comment:
We may have done this for other function, but I do not like it.  If we way this should be initialized to something, then this function should do it.  Otherwise, it should be an [in/out] parameter where we say that an entry is left untouched if the cache it not checked for that entry.

I know this is debatable, so I will not press it if you disagree.

Reply:
I agree it would be nicer. This is the same mechanism that is already in use elsewhere, so fixing this properly would require a larger change, which is probably out of scope of this PR (which is quite large already).